src/main/java/org/apache/accumulo/testing/performance/tests/GroupCommitPT.java

```diff
@@ -0,0 +1,234 @@
+        e.printStackTrace();
+      } finally {
+        try {
+          bw.close();
+        } catch (MutationsRejectedException e) {
+          e.printStackTrace();
+        }
+
+      }
+    }
+
+    long getTime() {
+      return time;
+    }
+  }
+
+  @Override
+  public SystemConfiguration getSystemConfig() {
+
+    Map<String,String> siteCfg = new HashMap<>();
+
+    siteCfg.put(Property.TSERV_MINTHREADS.getKey(), "128");
+    siteCfg.put(Property.TABLE_DURABILITY.getKey(), "sync");
+
+    return new SystemConfiguration().setAccumuloConfig(siteCfg);
+  }
+
+  @Override
+  public Report runTest(Environment env) throws Exception {
+
+    // int tests[] = new int[] {1, 2, 8, 16, 32, 128};
+    int numThreads = 128;
+
+    Report.Builder report = Report.builder();
+
+    report.id("mutslam");
+    report.description("Runs multiple threads to test performance of a group commit. "
+        + " This tests threads with client side group commit, using a single batch writer");
+
+    ArrayList<Object> batchValues = new ArrayList<>();
+
+    for (int i = 0; i < 6; i++) {
+      // This test threads w/ group commit on the client side, using a single batch writer.
+      // Each thread flushes after each mutation.
+      batchValues = runBatch(env, report, numThreads, 1);
+
+      report.info("threadsTime" + i,
+          new Double(new DecimalFormat("#0.00").format(batchValues.get(0))),
+          "Time it took the task to run in milliseconds");
+      report.info("threads" + i, i, "Number of threads");
+      report.info("batch" + i, 1, "Number of batches");
+      report.info("threadsMutations" + i,
+          new Double(new DecimalFormat("#0.00").format(batchValues.get(1))),
+          "Total number of mutations");
+      report.result("threadsRate" + i,
+          new Double(new DecimalFormat("#0.00").format(batchValues.get(2))),
+          "Total mutations per millisecond");
+    }
+
+    for (int i = 0; i < 6; i++) {
+      // This tests a single thread writing a different batch sizes of mutations,
+      // flushing after each batch. Group commit should approach these times for the same number
+      // mutations.
+      batchValues.clear();
+      batchValues = runBatch(env, report, 1, numThreads);
+
+      report.info("batchTime" + i,
+          new Double(new DecimalFormat("#0.00").format(batchValues.get(0))),
+          "Time it took the task to run in milliseconds");
+      report.info("threads" + i, i, "Number of threads");
+      report.info("batch" + i, 1, "Number of batches");
+      report.info("batchMutations" + i,
+          new Double(new DecimalFormat("#0.00").format(batchValues.get(1))),
+          "Total number of mutations");
+      report.result("batchRate" + i,
+          new Double(new DecimalFormat("#0.00").format(batchValues.get(2))),
+          "Total mutations per millisecond");
+    }
+
+    return report.build();
+  }
+
+  private ArrayList<Object> runBatch(Environment env, Report.Builder report, int numThreads,
```

Comment:
The report is passed in, but is it used?

Reply:
Nope, sure isn't.  Thanks for catching that.